app/Services/Deployment/FindViableNodesService.php

```diff
@@ -29,7 +29,7 @@ public function handle(int $memory = 0, int $disk = 0, int $cpu = 0, array $tags
             ->get();
 
         return $nodes
-            ->filter(fn (Node $node) => !$tags || collect($node->tags)->intersect($tags))
+            ->filter(fn (Node $node) => !$tags || !collect($node->tags)->intersect($tags)->isEmpty())
```

Comment:
```suggestion
            ->filter(fn (Node $node) => collect($node->tags)->intersect($tags)->isNotEmpty())
```
`$tags` can't be `null` or `false` since its an `array` even when not provided